test.js

```diff
@@ -136,3 +136,25 @@ test('support >2e9', (t) => {
   t.ok(compare(instance.max(), recordValue), 'max match')
   t.end()
 })
+
+test('reset histogram', (t) => {
+  const instance = new Histogram(1, 100)
+  t.equal(instance.min(), 9223372036854776000, 'min is setup')
```

Comment:
I've done some research on this, I don't believe it the issue we thought it. The min value should be initialised to positive infinity, and max value to negative infinity. It means the check for greater than or less than values are not complex logic trees, and you just use the single variables from initialisation in implementation.